app/controllers/navigation/Navigator.scala

```diff
@@ -52,22 +52,23 @@ class Navigator @Inject()(appConfig: AppConfig, auditService: AuditService) {
 
 object Navigator {
 
-  val standard: PartialFunction[DeclarationFieldCompanion, Mode => Call] = {
+  val standard: PartialFunction[DeclarationPage, Mode => Call] = {
     case BorderTransport => controllers.declaration.routes.DepartureTransportController.displayPage
-    case _               => ???
+    case Document        => controllers.declaration.routes.NatureOfTransactionController.displayPage
+    case _               => throw new IllegalStateException("Navigator back-link route not implemented")
   }
-
-  val supplementary: PartialFunction[DeclarationFieldCompanion, Mode => Call] = {
+  val supplementary: PartialFunction[DeclarationPage, Mode => Call] = {
     case BorderTransport => controllers.declaration.routes.DepartureTransportController.displayPage
-    case _               => ???
+    case Document        => controllers.declaration.routes.NatureOfTransactionController.displayPage
+    case _               => throw new IllegalStateException("Navigator back-link route not implemented")
   }
-
-  val simplified: PartialFunction[DeclarationFieldCompanion, Mode => Call] = {
+  val simplified: PartialFunction[DeclarationPage, Mode => Call] = {
     case BorderTransport => controllers.declaration.routes.WarehouseIdentificationController.displayPage
-    case _               => ???
+    case Document        => controllers.declaration.routes.OfficeOfExitController.displayPage
+    case _               => throw new IllegalStateException("Navigator back-link route not implemented")
```

Comment:
As I mentioned you I think `IllegalArgumentException` is more suitable for this, because the problem is related with wrong argument. Message is okay

Reply:
This has been addressed.